File: frontend/src/types/cluster.ts
export interface Cluster {
  id: string
  name: string
  description: string
  environment: 'dev' | 'test' | 'staging' | 'prod'
  provider: 'bare-metal' | 'vm' | 'cloud' | 'imported'
  version: string
  api_server: string
  status: 'healthy' | 'warning' | 'critical' | 'pending'
  created_at: string
  updated_at: string
}

export interface CreateClusterRequest {
  name: string
  description?: string
  environment: string
  provider: string
  version?: string
  api_server?: string
  kubeconfig?: string
}

export interface ClusterListResponse {
  clusters: Cluster[]
}

File: internal/cluster/cluster.go
package cluster

import "time"

type Cluster struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Environment string    `json:"environment"`
	Provider    string    `json:"provider"`
	Version     string    `json:"version"`
	APIServer   string    `json:"api_server"`
	Kubeconfig  string    `json:"-"` // Encrypted, never expose in JSON
	Status      string    `json:"status"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type ClusterRepository interface {
	Create(cluster *Cluster) error
	GetByID(id string) (*Cluster, error)
	GetByName(name string) (*Cluster, error)
	List() ([]*Cluster, error)
	ListByEnvironment(env string) ([]*Cluster, error)
	Update(cluster *Cluster) error
	Delete(id string) error
}
